runtime/fundamentals/testing.md

```diff
@@ -339,44 +339,107 @@ are up-to-date and functional.
  * # Examples
  *
  * ```ts
- * const x = 42;
+ * import { assertEquals } from "jsr:@std/assert/equals";
+ *
+ * const sum = add(1, 2);
+ * assertEquals(sum, 3);
  * ```
  */
+export function add(a: number, b: number): number {
+  return a + b;
+}
 ````
 
 The triple backticks mark the start and end of code blocks, the language is
-determined by the language identifier attribute which may be `js`, `jsx`, `ts`
-or `tsx`. If no language identifier is specified then the language is inferred
-from media type of the source document that the code block is extracted from.
+determined by the language identifier attribute which may be one of the
+following:
+
+- `js`
+- `javascript`
+- `mjs`
+- `cjs`
+- `jsx`
+- `ts`
+- `typescript`
+- `mts`
+- `cts`
+- `tsx`
+
+If no language identifier is specified then the language is inferred from media
+type of the source document that the code block is extracted from.
 
 ```sh
 deno test --doc example.ts
 ```
 
-The above command will extract this example, and then type-check it as a
-standalone module living in the same directory as the module being documented.
+The above command will extract this example, turn it into a pseudo test case
+that looks like below:
+
+```ts title="example.ts$4-10.ts" ignore
+import { assertEquals } from "jsr:@std/assert/equals";
+import { add } from "file:///path/to/example.ts";
+
+Deno.test("example.ts$4-10.ts", async () => {
+  const sum = add(1, 2);
+  assertEquals(sum, 3);
+});
+```
+
+and then run it as a standalone module living in the same directory as the
+module being documented.
+
+:::tip Want to type-check only?
+
+If you want to type-check your code snippets in JSDoc and markdown files without
+actually running them, you can use [`deno check`](/runtime/reference/cli/check/)
+command with `--doc` option (for JSDoc) or with `--doc-only` option (for
+markdown) instead.
+
+:::
 
-### Importing code in examples
+### Exported items are automatically imported
 
-You can import code from the surrounding module into the example code block by
-using the `import` keyword followed by the relative path to the module. This
-allows you to test the code in the example block in the context of the module
-being documented.
+Looking at the generated test code above carefully, you will notice that it
+includes the `import` statement to import the `add` fucntion even though the
+original code block does not have it. Briefly, all items imported from the
+module being documented are automatically imported into the generated test code
+with the same name.
```

Comment:
```suggestion
original code block does not have it. When documenting a module, any imported items are automatically included in the generated test code using with the same name.
```

Reply:
Oh I now realized that the original sentence was wrong. What I meant is "any exported items from the module will automatically get imported in the generated test code using the same name". I'll fix the sentence anyway following your suggestion 👍

Reply:
Fixed [610af29](https://github.com/denoland/docs/pull/860/commits/610af29aeb9a24a412ff07a14bbf7df4867c0dd6)